packages/ui/components/ui/tx/view/registry.tsx

```diff
@@ -0,0 +1,20 @@
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import {
+  AssetId,
+  Metadata,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+
+export function getStakingTokenMetaData(
+  chainId: string,
+  assetId: AssetId | undefined,
+): Metadata | undefined {
+  const registryClient = new ChainRegistryClient();
+  const registry = registryClient.get(chainId);
+
+  if (assetId === undefined) {
+    assetId = registry.stakingAssetId;
+  }
+  const StakingTokenMetadata = registry.getMetadata(assetId);
+
+  return StakingTokenMetadata;
+}
```

Comment:
It's hard to figure out 
What does this function do? Does it return metadata for the `assetId` you passed as a parameter, or metadata for the staking token if you passed undefined?
I think it would be better to do a little refactoring and split this logic into two functions

Reply:
I'll add some clarifying comments. After the TxV is generated, the `assetId` for the associated fee may either be `undefined` (meaning it uses the native staking token) or defined as the alt staking token. In the undefined case, we set the assetId to the staking token. We want to check for the undefined case because in transactions.tsx we're using the relative exponent, `const exponent = getDisplayDenomExponent(getStakingTokenMetaData(chain_id, asset_id))` associated with the asset metadata.

Reply:
Given that we can just change the planner, a good solution seems to be to have the planner always add the assetId to the fee.
But if we have reasons not to, I suggest renaming this function to `getFeeAssetMetadataOrDefault()` because the name `getStakingTokenMetaData()` may cause confusion with other functions that are named the same but have a different implementation